source-hubspot-native/source_hubspot_native/api.py

```diff
@@ -280,5 +352,171 @@ async def fetch_recent_tickets(
     return ((_ms_to_dt(r.timestamp), str(r.objectId)) for r in result), None
 
 
+# TODO(whb): As we add more object types that might need to use the search API for getting recent
+# changes, this function will probably make sense to generalize.
+async def fetch_recent_custom_objects(
+    object_name: str,
+    log: Logger,
+    http: HTTPSession,
+    since: datetime,
+    cursor: PageCursor,
+) -> tuple[Iterable[tuple[datetime, str]], PageCursor]:
+    
+    url = f"{HUB}/crm/v3/objects/{object_name}/search"
+
+    # The search API has known inconsistencies with very recent data.
+    horizon = datetime.now(tz=UTC) - CONSISTENCY_HORIZON_DELTA
+
+    if horizon <= since:
+        return [], None
```

Comment:
is it somehow possible to reach this condition?

Reply:
When the connector runs for the first time, `since` will be "now", and the horizon will be prior to that.